cypress/integration/DrugDictionaryAppintegration/DrugDictionaryAppintegration.js

```diff
@@ -0,0 +1,116 @@
+    });
+});
+
+Then('{string} option is displayed below the search bar', (browse) => {
+    cy.document().then(doc => {
+        const browseOption = doc.querySelector('span.browse');
+        expect(browseOption.firstChild.wholeText).to.include(browse);
+        expect(browseOption.parentNode.previousSibling.className).to.be.eq('drug-search');
+    })
+});
+
+Then('the page is showing the expand results for the letter {string}', (letter) => {
+    let results;
+
+    cy.url().then(url => {
+        results = `results found for: ${letter} `;
+        cy.get('#NCI-dd-app-root div.results h4').should('include.text', results);
+    });
+});
+
+And('user clicks on result number {int}', (resultNumber) => {
+    cy.get('div.results dfn').eq(resultNumber - 1).find('a').first().click();
+});
+
+Then('user is redirected to definition page and url contains {string}', (urlText) => {
+    cy.url().should('contain', urlText);
+});
+
+And('term title appears', () => {
+    cy.get('.dictionary-definiton__term-title').should('be.visible');
+});
+
+And('term definition appears', () => {
+    cy.get('div.dictionary-definiton__definition').should('be.visible');
+});
+
+And('the following links appear below definition text', () => {
+    cy.document().then(doc => {
+        const expectedLinks = doc.querySelector('div.dictionary-definiton__supplemental-links');
+        expect(expectedLinks.firstChild.className).eq('dictionary-definiton__active-trials-link');
+        expect(expectedLinks.lastChild.className).eq('dictionary-definiton__nci-thesaurus-link');
+        expect(expectedLinks.parentNode.className).to.be.eq('dictionary-definiton__term-definition');
+    })
+});
+
+When('user types {string} in the search bar', (textEntered) => {
+    cy.get('input[id="keywords"]').clear().type(textEntered);
+});
+
+Then('autosuggest appears', () => {
+    cy.get('div.ncids-autocomplete__menu.--terms[data-testid="tid-auto-suggest-options"]').should('be.visible');
+});
+
+
+When('user hits {string} key', (enterKey) => {
+    cy.get('input[id="keywords"]').type('{enter}{enter}');
+});
+
+Then('search is executed and page url is {string}', (urlText) => {
+    cy.location('href').then(url => {
+        expect(url.endsWith(urlText)).to.be.true;
+    });
+});
+
+Then('the no matching results page is displayed', () => {
+    cy.get('div.results p[data-testid="tid-no-matching-results"]').should('be.visible');
+});
+
+And('user selects {string} from the dropdown', (autosuggestTerm) => {
+    cy.get(".menu-wrapper div[class*='--terms']").should('be.visible');
+    cy.get(".menu-wrapper div[class*='--terms']").contains(autosuggestTerm).click();
+
+});
+
+And('user clears the search box', () => {
+    cy.get('input[id="keywords"]').clear();
+});
+
+And('user clicks {string} button', (searchButton) => {
+    cy.get('#btnSearch').trigger('click', { followRedirect: false });
+    cy.wait(500);
+});
+
+
```

Comment:
please remove empty lines after this (only one empty line is required after colon